src/renderers/dom/client/eventPlugins/TapEventPlugin.js

```diff
@@ -20,19 +21,59 @@ var ViewportMetrics = require('ViewportMetrics');
 var isStartish = EventPluginUtils.isStartish;
 var isEndish = EventPluginUtils.isEndish;
 
+import type {TopLevelTypes} from 'EventConstants';
+import type {ReactInstance} from 'ReactInstanceType';
+import type {PluginModule} from 'PluginModuleType';
+
+/**
+ * We are extending the Flow 'Touch' declaration to enable using bracket
+ * notation to access properties.
+ * Without this adjustment Flow throws
+ * "Indexable signature not found in Touch".
+ * See https://github.com/facebook/flow/issues/1323
+ */
+type TouchPropertyKey =
+  'clientX' |
+  'clientY' |
+  'pageX' |
+  'pageY';
+
+declare class _Touch extends Touch {
```

Comment:
Sounds like a good workaround, thanks for evaluating all the possible solutions :)